kafka/conn.py

```diff
@@ -366,35 +366,46 @@ def connect_blocking(self, timeout=float('inf')):
 
     def connect(self):
         """Attempt to connect and return ConnectionState"""
+        if self.config["socks5_proxy"] is not None and self._socks5_proxy is None:
+            self._socks5_proxy = Socks5Wrapper(self.config["socks5_proxy"], self.afi)
+
         if self.state is ConnectionStates.DISCONNECTED and not self.blacked_out():
             self.state = ConnectionStates.CONNECTING
             self.last_attempt = time.time()
-            next_lookup = self._next_afi_sockaddr()
-            if not next_lookup:
-                self.close(Errors.KafkaConnectionError('DNS failure'))
-                return self.state
-            else:
+            if self.config["socks5_proxy"] is None or not self._socks5_proxy.use_remote_lookup():
+                next_lookup = self._next_afi_sockaddr()
```

Comment:
Would it be simpler to move the `self._socks5_proxy.use_remote_lookup()` inside of `_next_afi_sockaddr()` ? For socks proxy w/ remote_lookup, it should return `(socket.AF_UNSPEC, (self.host, self.port))`. With that change I think most of these other changes would be unnecessary.

Reply:
Thanks, that indeed reduces the changes quite a bit! Let me know what you think of the current version!